File: hw5-zhiwei-zhiwei/oop_hw5/src/Printer.java
public class Printer implements IPrinter{
    private Cartridge cartridge;

    public Printer(Cartridge cartridge) {
        this.cartridge = cartridge;
    }

    public void prepare() {
        System.out.println("Printer is preparing to print");
        cartridge.useCartridge();
    }

    @Override
    public void print() {
        System.out.println("Printer is printing");
    }
}

File: hw5-zhiwei-zhiwei/hw5/Step.2/Printer.cpp
#include "Printer.h"
#include "Cartridge.h"

void Printer::prepare()
 {
	 cout << "Printer is preparing to print\n";
    string name = "X543-44";
    Cartridge * c = new Cartridge( name );
    delete c;
 }
void Printer::print()
 {
	 cout << "Printer is printing\n";
 }


